projects/packages/external-connections/src/class-external-connections.php

```diff
@@ -0,0 +1,272 @@
+<?php
+/**
+ * Management of external connections.
+ *
+ * @package automattic/jetpack-external-connections
+ */
+
+namespace Automattic\Jetpack;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Status\Host;
+
+/**
+ * Main class.
+ */
+class External_Connections {
+
+	const PACKAGE_VERSION = '0.1.0-alpha';
+	const BASE_FILE       = __FILE__;
+
+	/**
+	 * Gets the connect URL for a given service.
+	 *
+	 * @param string $service The service identifier.
+	 * @return string|null The connect URL, or `null` if the service is not supported.
+	 */
+	public static function get_connect_url( $service ) {
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			require_lib( 'external-connections' );
+			$connections = \WPCOM_External_Connections::init();
+			$service     = $connections->get_external_service_item( $service );
+			if ( ! empty( $service ) ) {
+				return $service['connect_URL'];
+			}
+		} else {
+			$site_id = Connection_Manager::get_site_id();
+			if ( is_wp_error( $site_id ) ) {
+				return null;
+			}
+
+			$path     = sprintf( '/sites/%d/external-services', $site_id );
+			$response = Client::wpcom_json_api_request_as_user( $path );
+			if ( is_wp_error( $response ) ) {
+				return null;
+			}
+
+			$body = json_decode( wp_remote_retrieve_body( $response ) );
+			if ( ! property_exists( $body, 'services' ) || ! property_exists( $body->services, $service ) ) {
```

Comment:
I think it could be replaced with isset($body->services->$service).

Better yet, it can be replaced directly with return $body->services->$service->connect_URL ?? null;